### PR TITLE
Use correct encoding for public secp256k1 key

### PR DESCRIPTION
**Pull request by jakubtrnka**
This is a suggested solution for  #721

### CODE REVIEW

File: utils/key-utils/src/lib.rs [hunk 1]
```diff
@@ -58,16 +58,31 @@ impl TryFrom<String> for Secp256k1PublicKey {
...
 
     fn try_from(value: String) -> Result<Self, Self::Error> {
-        let decoded = decode(value).into_vec()?;
-        let public = XOnlyPublicKey::from_slice(&decoded).expect("Invalid public key");
+        let decoded = decode(value).with_check(None).into_vec()?;
+        if decoded.len() < 34 {
+            return Err(Error::Custom);
+        }
+        if decoded[..2] != [1, 0] {
+            return Err(Error::Custom);
```
Comment (Sjors) [inline utils/key-utils/src/lib.rs:66]: Maybe something like `expect("Unsupported future version")`?

**Comment by Sjors** [comment]
Don't forgot to update the branch to main.

File: utils/key-utils/src/lib.rs [hunk 2]
```diff
@@ -29,45 +34,86 @@ impl From<secp256k1::Error> for Error {
...
 impl From<Secp256k1SecretKey> for String {
     fn from(secret: Secp256k1SecretKey) -> Self {
-        let bytes = secret.0.secret_bytes();
-        bs58::encode(bytes).into_string()
+        secret.to_string()
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+impl Display for Secp256k1SecretKey {
```
Comment (jakubtrnka) [inline utils/key-utils/src/lib.rs:65]: Display provides to_string method. See From impl

File: utils/key-utils/src/lib.rs (hunk 1 above)
Comment (jakubtrnka) [reply utils/key-utils/src/lib.rs]: done

**Comment by Fi3** [comment]
LGTM only cargo fmt and tests to fix, I will open a PR with the fixes against @jakubtrnka branch

**Comment by Fi3** [comment]
@jakubtrnka some tests are failing with InvalidChecksum:
```
 value: Bs58Decode(InvalidChecksum { checksum: [178, 200, 220, 184], expected_checksum: [9, 43, 108, 231] })
```

**Pull request merged**